interpreter/exec/eval.ml

```diff
@@ -132,6 +132,10 @@ let check_align addr ty sz at =
   if not (Memory.is_aligned addr ty sz) then
     Trap.error at "unaligned atomic memory access"
 
+let check_shared mem at =
+  if (shared_memory_type (Memory.type_of mem)) != Shared then
```

Comment:
```suggestion
  if shared_memory_type (Memory.type_of mem) <> Shared then
```
(`!=` is physical equality and almost never what you want for immutable values.)

Reply:
thanks!